### DISCUSSION TITLE
Seeking the Source Paper of the max filter

### DISCUSSION THREAD

**Discussion by ElonH** [halide/Halide#7711] [Q&A]
Hello, 
I'm trying to understand the fast max filter algorithm from 
https://github.com/halide/Halide/blob/main/apps/max_filter/max_filter_generator.cpp

It offers an incredibly efficient and elegant solution, which has piqued my interest.

However, it seems that this code lacks any references or information about its source paper. 
 
Now, I am eager to delve deeper into the background and principles behind this code, as well as understand the design philosophy and the research accomplishments of its creators. Hence, if anyone is aware of the source paper or any related information about this code, I kindly request you to leave a comment below to let me know.

**Comment by abadams** [comment]
I wrote that algorithm. I don't remember getting the idea from a paper or anything, it just seemed like a good way to do it.
